packages/slate/src/interfaces/editor.ts

```diff
@@ -1630,7 +1630,7 @@ export const Editor: EditorInterface = {
 
     for (const [node, path] of Editor.nodes(editor, {
       at: before,
-      match: Text.isText,
+      match: n => !Editor.isBlock(editor, n) && !Editor.isEditor(n),
```

Comment:
Might be clearer if it was specific in what it is including, e.g.

Text.isText || Editor.isVoid || Editor.isInline ?